# question 1

In [4]:
import scipy.stats as stats
import numpy as np

def variance_ratio_test(data1, data2):
    # Calculate the variances of the two samples
    var1 = np.var(data1, ddof=1)
    var2 = np.var(data2, ddof=1)
    
    # Calculate the F-value and p-value for the variance ratio test
    f_value = var1/var2
    p_value = stats.f.sf(f_value, len(data1)-1, len(data2)-1)
    
    return f_value, p_value

data1 = [1, 2, 3, 4, 5]
data2 = [2, 4, 6, 8, 10]
f_value, p_value = variance_ratio_test(data1, data2)
print("F-value:", f_value)
print("p-value:", p_value)

F-value: 0.25
p-value: 0.896


# question 2

In [5]:
import scipy.stats as stats

def critical_f_value(df_n, df_d, alpha=0.05):
    # Calculate the critical F-value for a two-tailed test
    f_crit = stats.f.isf(alpha/2, df_n, df_d)
    
    return f_crit

df_n = 3
df_d = 20
alpha = 0.05
f_crit = critical_f_value(df_n, df_d, alpha)
print("Critical F-value:", f_crit)

Critical F-value: 3.8586986662732143


# question 3

In [9]:
import numpy as np
import scipy.stats as stats

# Set the random seed for reproducibility
np.random.seed(123)

# Generate two random samples from normal distributions with known variances
n1 = 30
n2 = 30
mu1 = 0
mu2 = 0
var1 = 1
var2 = 1.5

x = np.random.normal(mu1, np.sqrt(var1), n1)
y = np.random.normal(mu2, np.sqrt(var2), n2)

# Calculate the F-value, degrees of freedom, and p-value for comparing variances
f_val = np.var(x, ddof=1) / np.var(y, ddof=1)
dfn = n1 - 1
dfd = n2 - 1
p_val = 2 * min(stats.f.cdf(f_val, dfn, dfd), 1 - stats.f.cdf(f_val, dfn, dfd))

# Print the results
print("F-value:", f_val)
print("Degrees of freedom (numerator, denominator):", dfn, dfd)
print("p-value:", p_val)


F-value: 0.6184089200351394
Degrees of freedom (numerator, denominator): 29 29
p-value: 0.20162963764723701


# question 4

To conduct an F-test for the equality of variances, we need to calculate the F-statistic as the ratio of the sample variances of the two populations. The null hypothesis is that the variances are equal, while the alternative hypothesis is that they are not equal.

The F-statistic is given by:

F = s1^2 / s2^2

where s1^2 and s2^2 are the sample variances of the first and second populations, respectively.

We need to compare the calculated F-value with the critical F-value from the F-distribution table. The degrees of freedom for the numerator and denominator are (n1-1) and (n2-1), respectively.

At a significance level of 5%, the critical F-value for (n1-1) = 11 and (n2-1) = 11 degrees of freedom is 3.11 (from the F-distribution table).

Let's calculate the F-statistic:

s1^2 = 10
s2^2 = 15
n1 = n2 = 12

F = s1^2 / s2^2 = 10 / 15 = 0.67

The calculated F-value (0.67) is less than the critical F-value (3.11) at the 5% significance level. Therefore, we fail to reject the null hypothesis and conclude that there is not enough evidence to suggest that the variances of the two populations are significantly different.

# question 5

To conduct an F-test for the variance of the diameter of a certain product, we need to calculate the F-statistic as the ratio of the sample variance to the claimed variance. The null hypothesis is that the claimed variance is true, while the alternative hypothesis is that it is not.

The F-statistic is given by:

F = s^2 / σ^2

where s^2 is the sample variance, and σ^2 is the claimed variance.

We need to compare the calculated F-value with the critical F-value from the F-distribution table. The degrees of freedom for the numerator and denominator are (n-1) and 1, respectively.

At a significance level of 1%, the critical F-value for (n-1) = 24 and df2 = 1 degrees of freedom is 6.60 (from the F-distribution table).

Let's calculate the F-statistic:

s^2 = 0.006
σ^2 = 0.005
n = 25

F = s^2 / σ^2 = 0.006 / 0.005 = 1.2

The calculated F-value (1.2) is less than the critical F-value (6.60) at the 1% significance level. Therefore, we fail to reject the null hypothesis and conclude that there is not enough evidence to suggest that the claimed variance of the diameter of a certain product is not justified.

# question 6


In [5]:
import math

def f_distribution_mean_var(df1, df2):
    if df1 <= 0 or df2 <= 0:
        raise ValueError("Degrees of freedom must be positive")
    
    if df2 == 1:
        return (math.inf, math.inf)
    
    mean = df2 / (df2 - 2)
    var = (2 * (df2**2) * (df1 + df2 - 2)) / ((df1 * (df2 - 2)**2 * (df2 - 4)))
    
    return (mean, var)


f_distribution_mean_var(5, 10)

(1.25, 1.3541666666666667)

# question 7

To conduct an F-test for the equality of variances between two populations, we need to calculate the test statistic F, which is the ratio of the sample variances, and compare it to the F-distribution with (n1-1) and (n2-1) degrees of freedom, where n1 and n2 are the sample sizes of the two populations.

The null hypothesis is that the variances of the two populations are equal, and the alternative hypothesis is that they are not equal.

Given the sample variances s1^2 = 25 and s2^2 = 20, the test statistic F is:

F = s1^2 / s2^2 = 25 / 20 = 1.25

The degrees of freedom for the F-distribution are (10-1) = 9 and (15-1) = 14, respectively.

Using a statistical software or a F-distribution table, we can find the critical F-value for a 10% significance level and (9,14) degrees of freedom, which is approximately 2.34.

Since the calculated F-value (1.25) is less than the critical F-value (2.34), we fail to reject the null hypothesis and conclude that there is not enough evidence to suggest that the variances of the two populations are significantly different at the 10% significance level.

Therefore, we cannot conclude that the variances are significantly different between the two populations.

# question 8

To conduct an F-test to determine if the variances of the waiting times at Restaurant A and Restaurant B are significantly different at the 5% significance level, we need to follow these steps:

Step 1: Calculate the variance and sample size for each restaurant. We will use the formula for variance:

s^2 = Σ(x - x̄)^2 / (n - 1)

where s^2 is the sample variance, x is the individual data point, x̄ is the sample mean, and n is the sample size.

For Restaurant A:

n = 7

x̄ = (24+25+28+23+22+20+27)/7 = 24.14

s^2 = [ (24-24.14)^2 + (25-24.14)^2 + (28-24.14)^2 + (23-24.14)^2 + (22-24.14)^2 + (20-24.14)^2 + (27-24.14)^2 ] / (7-1) = 10.26

For Restaurant B:

n = 6

x̄ = (31+33+35+30+32+36)/6 = 32.83

s^2 = [ (31-32.83)^2 + (33-32.83)^2 + (35-32.83)^2 + (30-32.83)^2 + (32-32.83)^2 + (36-32.83)^2 ] / (6-1) = 6.94

Step 2: Calculate the F statistic using the formula:

F = s1^2 / s2^2

where s1^2 is the variance of Restaurant A and s2^2 is the variance of Restaurant B.

F = 10.26 / 6.94 = 1.48

Step 3: Find the critical F-value using a table or calculator. Since we have a sample size of 6 and 7 degrees of freedom (df), we can find the critical value using an F-table or calculator with alpha=0.05 and degrees of freedom 6 and 7. The critical F-value is 4.75.

Step 4: Compare the F-statistic to the critical F-value. Since 1.48 is less than 4.75, we fail to reject the null hypothesis that the variances are equal. This means that we do not have enough evidence to conclude that the variances of the waiting times at the two restaurants are significantly different at the 5% significance level.

Therefore, we can conclude that there is not enough evidence to suggest that the variance of waiting times is significantly different between Restaurant A and Restaurant B at the 5% significance level.





# question 9

To conduct an F-test to determine if the variances of the test scores of Group A and Group B are significantly different at the 1% significance level, we need to follow these steps:

Step 1: Calculate the variance and sample size for each group. We will use the formula for variance:

s^2 = Σ(x - x̄)^2 / (n - 1)

where s^2 is the sample variance, x is the individual data point, x̄ is the sample mean, and n is the sample size.

For Group A:
    
n = 6

x̄ = (80+85+90+92+87+83)/6 = 86.17

s^2 = [ (80-86.17)^2 + (85-86.17)^2 + (90-86.17)^2 + (92-86.17)^2 + (87-86.17)^2 + (83-86.17)^2 ] / (6-1) = 31.97

For Group B:
    
n = 6

x̄ = (75+78+82+79+81+84)/6 = 80.17

s^2 = [ (75-80.17)^2 + (78-80.17)^2 + (82-80.17)^2 + (79-80.17)^2 + (81-80.17)^2 + (84-80.17)^2 ] / (6-1) = 11.97

Step 2: Calculate the F statistic using the formula:

F = s1^2 / s2^2

where s1^2 is the variance of Group A and s2^2 is the variance of Group B.

F = 31.97 / 11.97 = 2.67

Step 3: Find the critical F-value using a table or calculator. Since we have a sample size of 6 and 6 degrees of freedom (df), we can find the critical value using an F-table or calculator with alpha=0.01 and degrees of freedom 6 and 6. The critical F-value is 6.39.

Step 4: Compare the F-statistic to the critical F-value. Since 2.67 is less than 6.39, we fail to reject the null hypothesis that the variances are equal. This means that we do not have enough evidence to conclude that the variances of the test scores of the two groups are significantly different at the 1% significance level.

Therefore, we can conclude that there is not enough evidence to suggest that the variance of test scores is significantly different between Group A and Group B at the 1% significance level.





